File: diet/Herbivore.cpp
/* NIM/Nama  : 13515078 / Veren Iliana K   */
/* File      : Herbivore.cpp               */

#include "Herbivore.h"

Herbivore::Herbivore() : meatRatio(0)
{
	vegRatio = 0.3;
}

Herbivore::Herbivore(const Herbivore& H) : meatRatio(C.meatRatio)
{
	vegRatio = C.meat;
}

Herbivore& Herbivore::operator= (const Herbivore& H)
{
	meatRatio = 0;
	vegRatio = 0.3;
	return *this;
}

File: Herbivore.java
/**
 * @author NIM/Nama : 13515078 / Veren Iliana Kurniadi
 * File : Herbivore.java
 */

import java.util.*;
import java.lang.*;

public interface Herbivore {
  /**
   * Mengembalikan nilai rasio daging.
   * @return nilai rasio daging
   */
  public double getMeatRatio();

  /**
   * Mengembalikan nilai rasio sayur.
   * @return nilai rasio sayur
   */
  public double getVegRatio();
}
